tests/FileReaderTest.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Tests;
+
+use org\bovigo\vfs\vfsStream;
+use PhpCsFixer\FileReader;
+
+/**
+ * @author ntzm
+ *
+ * @internal
+ *
+ * @covers \PhpCsFixer\FileReader
+ */
+final class FileReaderTest extends TestCase
+{
+    public function testCreateSingleton()
+    {
+        $instance = FileReader::createSingleton();
+
+        $this->assertSame($instance, FileReader::createSingleton());
+    }
+
+    public function testRead()
+    {
+        $fs = vfsStream::setup('root', null, array(
+            'foo.php' => '<?php echo "hi";',
+        ));
+
+        $reader = new FileReader();
+
+        $this->assertSame('<?php echo "hi";', $reader->read($fs->url().'/foo.php'));
+    }
+
+    public function testReadStdinCaches()
+    {
+        $reader = new FileReader();
+
+        stream_wrapper_unregister('php');
+        stream_wrapper_register('php', 'PhpCsFixer\Tests\Fixtures\Test\FileReaderTest\StdinFakeStream');
+
+        $this->assertSame('<?php echo "foo";', $reader->read('php://stdin'));
+        $this->assertSame('<?php echo "foo";', $reader->read('php://stdin'));
+
+        stream_wrapper_restore('php');
```

Comment:
This should be done in `tearDownAfterClass()` in case an assertion fails (i.e. throws an exception).